language/stdlib/modules/Roles.move

```diff
@@ -179,14 +179,15 @@ module Roles {
     spec fun grant_role {
         pragma opaque;
         include GrantRole;
+        let addr = Signer::spec_address_of(account);
+        // Requires to satisfy global invariants.
+        requires role_id == LIBRA_ROOT_ROLE_ID ==> addr == CoreAddresses::LIBRA_ROOT_ADDRESS();
```

Comment:
I think this should be assertions in the code and aborts_if in the specs.

Reply:
As I understand this comment, you're suggesting that I add an assert to grant_role that role_id is libra root.  But this seems to me exactly the kind of situation where we would want to use "requires".

grant_role is private and every calling context has an assert that the role_id is libra root.  So, this seemed right to me, since it's checked in all calling contexts and helps to prove the invariant.